src/mame/drivers/spec128.cpp

```diff
@@ -198,8 +198,8 @@ u8 spectrum_128_state::spectrum_128_rom_r(offs_t offset)
 		: ((u8*)m_bank_rom[0]->base())[offset];
 }
 
-template <u8 Bank>
-void spectrum_128_state::spectrum_128_ram_w(offs_t offset, u8 data)
+template void spectrum_128_state::spectrum_128_ram_w<0>(offs_t offset, u8 data);
+template <u8 Bank> void spectrum_128_state::spectrum_128_ram_w(offs_t offset, u8 data)
```

Comment:
This may not work with some compilers – at least some versions of clang require the definitions to be available at the point of the explicit instantiation.  The explicit instantiation will need to come *after* the function body.  Also, are you missing an explicit instantiation for `spectrum_128_state::spectrum_128_ram_r<0>`?